components/crm/internal/bootstrap/streaming.go

```
@@ -0,0 +1,299 @@
+//   - When STREAMING_BROKERS is empty (LoadConfig surfaces this as an empty
+//     Brokers slice) the function ALSO returns a NoopEmitter — the Builder
+//     would otherwise reject construction with a missing-target error.
+//   - When no events are registered (crmEventDefinitions() is empty) the
+//     function ALSO returns a NoopEmitter. The Builder validates empty ROUTES
+//     first (NewRouteTable rejects a no-routes-configured table before the
+//     catalog check is reached), so an empty catalog would crash bootstrap.
+//   - Otherwise the function builds a single-target catalog-first Producer via
+//     libStreaming.NewBuilder(), wiring the configured CloudEvents source onto
+//     the Builder and registering all CRM event definitions in the Catalog
+//     with a matching RouteDefinition per event.
+func BuildStreamingEmitter(
```

golangci-lint (2.12.2)

[error] cyclomatic complexity 19 of func `BuildStreamingEmitter` is high (> 18)

(gocyclo)